liquidctl/driver/asetek_pro.py

```diff
@@ -0,0 +1,271 @@
+            LOGGER.debug('color %s', color)
+            colorCount = colorCount + 1
+            setColors = setColors + color
+        self._write([0x56, colorCount] + setColors)
+        self._end_transaction_and_read(3)
+
+    def _get_fan_speeds(self):
+        """Read the RPM speed of the fans. This Tries to get the speeds for 3 fans as that is
+        how many the H150i PRO has"""
+        speeds = []
+        for i in [0, 1, 2]:
+            self._write([_PRO_CMD_READ_FAN_SPEED, i])
+            msg = self._end_transaction_and_read(6)
+            if msg[0] != 0x41 or msg[1] != 0x12 or msg[2] != 0x34 or msg[3] != i:
+                LOGGER.debug('Unable to get speed for fan id {}'.format(i))
+                continue
+            speeds.append((msg[4] << 8) + msg[5])
+        return speeds
+
+    def _get_fan_indexes(self, channel):
+        if len(channel) > 3:
+            return [int(channel[3:]) - 1]
+        return range(len(self._get_fan_speeds()))
+
+    def initialize(self, pump_mode=None, **kwargs):
+        """Initialize the device."""
+        LOGGER.debug('Pro configure device...')
+        if pump_mode == None:
+            return
+        else:
+            LOGGER.debug('pump_mode %s write value %s', pump_mode,
+                format(_PRO_PUMP_MODES.index(pump_mode),'02x'))
+            self._write([_PRO_CMD_WRITE_PUMP_MODE, _PRO_PUMP_MODES.index(pump_mode)])
+            self._end_transaction_and_read(5)
+
+    def get_status(self, **kwargs):
+        """Get a status report.
+        Returns a list of `(property, value, unit)` tuples.
+        """
+        status = []
+        ### Can some Pro devices have more than one temp sensor?   OCL seems to code for that?
+        self._write([_PRO_CMD_READ_AIO_TEMP])
+        msg = self._end_transaction_and_read(6)
+        #LOGGER.debug('aio temp msg %s', ' '.join(format(i, '02x') for i in msg))
+        aio_temp = msg[3] + msg[4]/10
+        status.append(('Liquid temperature', aio_temp, '°C'))
+        speeds = self._get_fan_speeds()
+        for i in range(len(speeds)):
+            status.append(('Fan {} speed'.format(i+1), speeds[i] | 0, 'rpm'))
+        self._write([_PRO_CMD_READ_PUMP_MODE])
+        msg = self._end_transaction_and_read(4)
+        #LOGGER.debug('pump mode msg %s', ' '.join(format(i, '02x') for i in msg))
+        pump_mode = msg[3]
+        status.append(('Pump mode', _PRO_PUMP_MODES[pump_mode], ""))
+        self._write([_PRO_CMD_READ_PUMP_SPEED])
+        msg = self._end_transaction_and_read(5)
+        #LOGGER.debug('pump speed msg %s', ' '.join(format(i, '02x') for i in msg))
+        pump_speed = (msg[3] << 8) + msg[4]
+        status.append(('Pump speed', pump_speed | 0, 'rpm'))
+        self._write([_PRO_CMD_READ_FIRMWARE])
+        msg = self._end_transaction_and_read(7)
+        #LOGGER.debug('firmware id msg %s', ' '.join(format(i, '02x') for i in msg))
+        firmware = '{}.{}.{}.{}'.format(format(msg[3], '01x'), format(msg[4], '01x'),
```

Comment:
see for an example of a potentially cleaner way of writing the firmware version string from one of the other drivers:

https://github.com/jonasmalacofilho/liquidctl/blob/90d7ec50d7235c58b6ee0d16e602763ed2a1bb9a/liquidctl/driver/hydro_platinum.py#L175-L178

Reply:
Done